src/pages/groupComparison/groupSelector/GroupSelector.tsx

```diff
@@ -206,6 +206,15 @@ export default class GroupSelector extends React.Component<
                         index={buttons.length}
                     />
                 );
+                // Logic To Put Primary First
```

Comment:
i think we need to extract the logic here (163 onward) and put it in a testable function.  also we shouldn't be "editing" the props of a group.  we need to use functional approach and produce a new ordered data (or do ordering upstream)

Reply:
True that, allow me some time to work on this. Thanks!